### DISCUSSION TITLE
I just want to know about usage of  _creation_counter in Field class? and why it is there?

### DISCUSSION THREAD

**Discussion by matrosdzmehedi** [encode/django-rest-framework#9519] [Question & Answer] [answered]
class Field:
    _creation_counter = 0

    default_error_messages = {
        'required': _('This field is required.'),
        'null': _('This field may not be null.')
    }
    default_validators = []
    default_empty_html = empty
    initial = None

    def __init__(self, *, read_only=False, write_only=False,
                 required=None, default=empty, initial=empty, source=None,
                 label=None, help_text=None, style=None,
                 error_messages=None, validators=None, allow_null=False):
        self._creation_counter = Field._creation_counter
        Field._creation_counter += 1

**Comment by lovelydinosaur** [comment, marked as answer]
It was originally there as it was required in order to track the ordering on serializer fields. Presumably no longer required and can be removed?

**Reply by matrosdzmehedi** [reply]
Is this for ordering or just counting fields that has been created?

**Reply by lovelydinosaur** [reply]
It was for ordering.

**Reply by TGoddessana** [reply]
https://github.com/encode/django-rest-framework/blob/b25028ac8f46984459f9d9e28069b797986ae476/rest_framework/serializers.py#L288C9-L288C58

That private field seems to be used here; the entire test passes even without it(the `fields.sort(key=lambda x: x[1]._creation_counter)` line)

**Reply by lovelydinosaur** [reply]
Thanks @TGoddessana okay good to know it's still used. It's possible(?) that it's redundant since Python switched to ordered dictionaries, but given that it's in use there perhaps let's just leave it and not introduce unnecessary risk?
